File: app/src/androidTest/java/pavelnazimok/uitestingfeatures/java/robots/FirstScreenRobot.java
package pavelnazimok.uitestingfeatures.java.robots;

import pavelnazimok.uitestingfeatures.R;

import static com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertDisplayed;
import static com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn;
import static pavelnazimok.uitestingfeatures.java.utils.EspressoExtensions.getText;
import static pavelnazimok.uitestingfeatures.java.utils.EspressoExtensions.waitForView;

@SuppressWarnings("UnusedReturnValue")
public class FirstScreenRobot {

    public int getCoffeePrice() {
        return Integer.valueOf(getText(R.id.label_coffee_price));
    }

    public FirstScreenRobot clickCoffeeIncrement() {
        clickOn(R.id.button_coffee_increment);
        return this;
    }

    public FirstScreenRobot clickCoffeeDecrement() {
        clickOn(R.id.button_coffee_decrement);
        return this;
    }

    public FirstScreenRobot assertFirstScreenIsDisplayed() {
        waitForView(R.id.screen_first, 5000);
        return this;
    }

    public FirstScreenRobot assertCoffeeCountEquals(final int count) {
        assertDisplayed(R.id.label_coffee_count, String.valueOf(count));
        return this;
    }

    public FirstScreenRobot assertCoffeePriceEquals(final int price) {
        assertDisplayed(R.id.label_coffee_price, String.valueOf(price));
        return this;
    }

    public FirstScreenRobot assertTotalPriceEquals(final int price) {
        assertDisplayed(R.id.label_total_price, String.valueOf(price));
        return this;
    }
}

File: app/src/androidTest/java/pavelnazimok/uitestingfeatures/kotlin/robots/FirstScreenRobot.kt
package pavelnazimok.uitestingfeatures.kotlin.robots

import com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertDisplayed
import com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn
import pavelnazimok.uitestingfeatures.R
import pavelnazimok.uitestingfeatures.kotlin.utils.getText
import pavelnazimok.uitestingfeatures.kotlin.utils.waitForView

fun onFirstScreen(func: FirstScreenRobot.() -> Unit): FirstScreenRobot {
    return FirstScreenRobot().apply {
        assertFirstScreenIsDisplayed()
        func()
    }
}

class FirstScreenRobot {

    fun getCoffeePrice(): Int {
        return Integer.valueOf(getText(R.id.label_coffee_price))
    }

    fun clickCoffeeIncrement() {
        clickOn(R.id.button_coffee_increment)
    }

    fun clickCoffeeDecrement() {
        clickOn(R.id.button_coffee_decrement)
    }

    fun assertFirstScreenIsDisplayed() {
        waitForView(R.id.screen_first)
    }

    fun assertCoffeeCountEquals(count: Int) {
        assertDisplayed(R.id.label_coffee_count, count.toString())
    }

    fun assertCoffeePriceEquals(price: Int) {
        assertDisplayed(R.id.label_coffee_price, price.toString())
    }

    fun assertTotalPriceEquals(price: Int) {
        assertDisplayed(R.id.label_total_price, price.toString())
    }
}
